woocommerce/pinterest-for-woocommerce
# Dynamically generate amount required to spend for credits to be given based on currency

### Changes proposed in this Pull Request:

We display a static message in the ads credits notice (at multiple locations) that states a merchant can get $125 credits on spending $15. We want to dynamically change the credits based on the merchant's store currency and display a custom notice. 

This PR changes the credit values in the Ads credits notice based on store currency. Merchants will see credits in their respective currency if it belongs to EUR, GBP, BRL, CAD, AUD, MXN, ARS, CHF, CZK, DKK, HUF, JPY, NOK, NZD, PLN, RON, or SEK. For all other currencies, including USD, they will see notice in USD credits.

`get_currency_credits()` function returns spend requirement and credits given values based on store currency. `add_currency_credits_info_to_account_data()` function appends the credits info to account data we get from Pinterest. We are able to fetch the values from `account_data` and display them in the notice.

The credits question in FAQ had the same `questionId` as the previous one (multiple accounts question) so this PR changes it to a relevant value `'how-to-redeem-ad-credits'`. 

Partially fixes #801.

### Screenshots:

| Before | After |
|--------|--------|
| <img width="1033" alt="Screenshot 2023-08-18 at 9 00 01 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/21baf646-1809-43c4-a5c7-e8e99f9407ba"> | <img width="1037" alt="Screenshot 2023-08-21 at 9 28 15 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/04af7553-be7e-45fc-bdb3-ce9d04fe4d4b"> |
| <img width="368" alt="AdsCreditsPromo" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/6df30d55-adba-4af5-8755-2fb4c5b72c76"> | <img width="473" alt="Screenshot 2023-08-21 at 9 23 06 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/7606f738-3ae0-4aaa-ae56-444e69ae3c7c"> |
| <img width="492" alt="Screenshot 2023-08-18 at 8 59 08 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/59123813-b2dd-4618-a81f-8f7fdc3f8b8f"> | <img width="511" alt="Screenshot 2023-08-21 at 9 28 08 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/175ee132-bfb2-46a3-aaff-ad2db90c7ff5"> |
| <img width="849" alt="Screenshot 2023-08-21 at 5 04 52 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/add55b78-aa12-4071-b857-5a194668fd7b"> | <img width="840" alt="Screenshot 2023-08-21 at 9 21 39 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/4f5364e8-ce38-4f17-b71c-c0a5619ced09"> |
| <img width="688" alt="Screenshot 2023-08-21 at 5 04 58 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/6d934f73-91e4-4c77-8624-502de1bae125"> | <img width="725" alt="Screenshot 2023-08-21 at 9 22 46 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/37e9f195-423b-4b4c-8dee-6bbe8c165383"> | 
| <img width="1098" alt="Screenshot 2023-08-21 at 5 05 06 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/a927cfc3-b765-4670-8bb7-71e17b5d062d"> | <img width="981" alt="Screenshot 2023-08-21 at 9 23 14 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/c694bccf-4beb-4417-8255-afddcb3f6842"> |
| <img width="833" alt="Screenshot 2023-08-21 at 9 22 19 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/4eaca002-c572-422c-9fde-ef40fbd0e423"> | <img width="825" alt="Screenshot 2023-08-21 at 9 22 38 PM" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/33723519/e55a2e1a-ddb6-4e40-8f41-d5b23ae06cc8"> | 

### Detailed test instructions:


1. Clone the repo and checkout this branch 
2. Activate the plugin
3. Set currency to Euros at WooCommerce > Settings > Currency Options > Currency
4. Go to Marketing > Pinterest
5. Verify you see all credits in euros in "Try Pinterest for WooCommerce and get €131 in ad credits!" notice
6. Click on Pinterest Terms and Conditions link under it and verify credits are in euros
7. Scroll down to FAQ section
8. Verify credits are in euros in both question and answer
9. Click on Get Started
10. On the onboarding page, verify the ads credits promo shows credits in euros
11. Click on Pinterest Terms and Conditions link under it and verify credits are in euros
12. Complete the onboarding process and connect your business account
13. Go to Marketing > Pinterest > Catalog
14. Verify credit values are in euros in the onboarding pop up modal
15. Close the modal and verify you see credits in euros on Ads credits notice
16. Repeat steps 4 to 15 with different currencies like USD, GBP, CZK, MXN


### Additional details:


### Changelog entry

> Add - Adds logic to dynamically display spend requirement and credits given based on store currency.
